core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/cloner/FieldAccessingSolutionCloner.java

```diff
@@ -257,6 +226,10 @@ private static <K, V> Map<K, V> constructCloneMap(Map<K, V> originalMap) {
         }
     }
 
+    private ClassMetadata retrieveClassMetadata(Class<?> declaringClass) {
+        return classMetadataMemoization.computeIfAbsent(declaringClass, ClassMetadata::new);
+    }
```

Comment:
Putting all of the decisions into a single `ClassMetadata` class allows us to only query the `ConcurrentMap` once instead of three times. This turns out to be very important, as the original code would spend 25 % of solver runtime here. (machinereassignment B-7 with SA.)